### PR TITLE
Lint influxql/nedelmead package

### PR DESCRIPTION
**Pull request by d2bit**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Part of #4098.

### CODE REVIEW

File: influxql/neldermead/neldermead.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ const (
 	defaultGamma = 2.0
 )
 
+// Optimizer represents the neder-mead algorithm configuration.
```
Comment (jwilder) [inline influxql/neldermead/neldermead.go]: Nelder-Mead

File: influxql/neldermead/neldermead.go [hunk 2]
```diff
@@ -33,6 +35,7 @@ func New() *Optimizer {
 	}
 }
 
+// Optimize returns the neder-mead algorith result.
```
Comment (jwilder) [inline influxql/neldermead/neldermead.go]: Nelder-Mead

Comment (jwilder) [inline influxql/neldermead/neldermead.go]: Also `algorithm`.

**Comment by e-dard** [comment]
@d2bit could you address the feedback or we'll have to close this PR.

**Pull request closed without merging**